addons/html_editor/static/src/main/font/font_plugin.js

```diff
@@ -308,4 +310,35 @@ export class FontPlugin extends Plugin {
         this.shared.setCursorStart(p);
         return true;
     }
+
+    onInput(ev) {
+        if (ev.data !== ' ') {
+            return;
+        }
+        const selection = this.shared.getEditableSelection();
+        const blockEl = closestBlock(selection.anchorNode);
+        const leftDOMPath = leftLeafOnlyNotBlockPath(selection.anchorNode);
+        let spaceOffset = selection.anchorOffset;
+        let leftLeaf = leftDOMPath.next().value;
+        while (leftLeaf) {
+            // Find offset if there is a situation like this:
+            // <p>ab<strong>cd []e</strong></p>
+            spaceOffset += leftLeaf.length;
+            leftLeaf = leftDOMPath.next().value;
+        }
```

Comment:
Could you add a test case for this scenario. Also, since the same issue might occur with automatic numbered list detection, could you fix that in a new commit as well?